### ISSUE TITLE
New task: datum hash -> datum table

### ISSUE THREAD

**Issue by SebastienGllmt** [dcSpark/carp#80]
Lots of people want an indexer capable of mapping datum hash to datum. We should be able to create a task for this and then expose it in the server/client after

**Comment by MitchTurner** [comment]
Started doing some TDD for this and found that it's painful constructing some of the Pallas types (e.g. `TransactionBody`)  the tasks depend on directly. It looks like the `main` branch of Pallas makes the fields of the structs `pub` (at least for `TransactionBody`) , so it should be easier to construct in the future. Otherwise I might need to decode from CBOR directly in the test helpers :(.

It would be nice if carp had its own domain types for all the primitives rather than depending directly on Pallas. It's always a big undertaking up front to do that, but it would help avoid problems like this.

**Comment by MitchTurner** [comment]
Yeah. `pallas v0.11.0-alpha.1` has the needed changes. I would recommend waiting until carp/tasks is updated to that version before trying to right the tests for the datum task. It's a non-trivial amount of work to update.

**Comment by SebastienGllmt** [comment]
Ideally we would deprecate all of Pallas

**Comment by MitchTurner** [comment]
What do you mean by "deprecate"? Swap them out with domain types?

**Comment by SebastienGllmt** [comment]
Swap it all out with CML once all the CML upgrades are in place. Cardano Multiplatform Lib still has some limitations which means we currently have to use both, but that's kind of awkward and not ideal.
